MapboxNavigation/RouteManeuverViewController.swift

```diff
@@ -1,25 +1,75 @@
+    
     weak var step: RouteStep!
     
+    public func notifyDidChange(routeProgress: RouteProgress, secondsRemaining: TimeInterval) {
+        let stepProgress = routeProgress.currentLegProgress.currentStepProgress
+        let distanceRemaining = stepProgress.distanceRemaining
+        
+        distance = secondsRemaining > 5 ? distanceRemaining : nil
+        
+        if routeProgress.currentLegProgress.alertUserLevel == .arrive {
+            distance = nil
+            streetLabel.text = routeStepFormatter.string(for: routeStepFormatter.string(for: routeProgress.currentLegProgress.upComingStep))?.abbreviated(toFit: availableStreetLabelBounds, font: streetLabel.font)
+        } else if let upComingStep = routeProgress.currentLegProgress?.upComingStep {
+            if let name = upComingStep.names?.first {
+                streetLabel.text = name.abbreviated(toFit: availableStreetLabelBounds, font: streetLabel.font!)
+            } else if let destinations = upComingStep.destinations?.joined(separator: "\n") {
+                streetLabel.text = destinations.abbreviated(toFit: availableStreetLabelBounds, font: streetLabel.font!)
+            } else {
+                streetLabel.text = routeStepFormatter.string(for: upComingStep)?.abbreviated(toFit: availableStreetLabelBounds, font: streetLabel.font)
+            }
+            
+            // TODO: Factor out and recalculate bounds after shield has been updated
+            //updateShield(for: controller)
+            
+            showLaneView(step: upComingStep)
+            
+            // TODO: fix animation
+            //            if !controller.isPagingThroughStepList {
+            //                let initialPaddingForOverviewButton:CGFloat = controller.stackViewContainer.isHidden ? -30 : -20 + controller.laneViews.first!.frame.maxY
+            //                UIView.animate(withDuration: 0.5, animations: {
+            //                    self.overviewButtonTopConstraint.constant = initialPaddingForOverviewButton + controller.stackViewContainer.frame.maxY
+            //                })
+            //            }
+        }
+        
+        turnArrowView.step = routeProgress.currentLegProgress.upComingStep
+    }
+    
+    public func updateStreetNameForStep() {
+        if let name = step?.names?.first {
+            streetLabel.text = name.abbreviated(toFit: availableStreetLabelBounds, font: streetLabel.font)
+        } else if let destinations = step?.destinations?.joined(separator: "\n") {
+            streetLabel.text = destinations.abbreviated(toFit: availableStreetLabelBounds, font: streetLabel.font)
+        } else if let step = step {
+            streetLabel.text = routeStepFormatter.string(for: step)?.abbreviated(toFit: availableStreetLabelBounds, font: streetLabel.font)
+        }
+    }
+    
     var distance: CLLocationDistance? {
         didSet {
             if let distance = distance {
+                distanceLabel.isHidden = false
```

Comment:
Usually, but I think it also depends on if it's contained in a stack view and what the distribution is set to. The width will shrink to 0 when the text is nil but height will still be based on the font size.